receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go

```diff
@@ -137,29 +119,56 @@ func initializeFileSystemUsageMetric(metric pdata.Metric, now pdata.TimestampUni
 	}
 }
 
-func initializeFileSystemUsageDataPoint(dataPoint pdata.IntDataPoint, now pdata.TimestampUnixNano, deviceLabel string, stateLabel string, value int64) {
+func initializeFileSystemUsageDataPoint(dataPoint pdata.IntDataPoint, now pdata.TimestampUnixNano, partition disk.PartitionStat, stateLabel string, value int64) {
 	labelsMap := dataPoint.LabelsMap()
-	labelsMap.Insert(deviceLabelName, deviceLabel)
+	labelsMap.Insert(deviceLabelName, partition.Device)
+	labelsMap.Insert(typeLabelName, partition.Fstype)
+	labelsMap.Insert(mountModeLabelName, getMountMode(partition.Opts))
+	labelsMap.Insert(mountPointLabelName, partition.Mountpoint)
 	labelsMap.Insert(stateLabelName, stateLabel)
 	dataPoint.SetTimestamp(now)
 	dataPoint.SetValue(value)
 }
 
-func (s *scraper) filterByDevice(usages []*deviceUsage) []*deviceUsage {
-	if s.includeFS == nil && s.excludeFS == nil {
-		return usages
+func getMountMode(opts string) string {
+	splitOptions := strings.Split(opts, ",")
+	if exists(splitOptions, "rw") {
+		return "rw"
+	} else if exists(splitOptions, "ro") {
+		return "ro"
 	}
+	return "unknown"
+}
 
-	filteredUsages := make([]*deviceUsage, 0, len(usages))
-	for _, usage := range usages {
-		if s.includeDevice(usage.deviceName) {
-			filteredUsages = append(filteredUsages, usage)
+func exists(options []string, opt string) bool {
+	for _, o := range options {
+		if o == opt {
+			return true
 		}
 	}
-	return filteredUsages
+	return false
+}
+
+func (f *fsFilter) includePartition(partition disk.PartitionStat) bool {
```

Comment:
It's probably worth keeping the optimization for the simple case where no filters are applied, i.e. create a flag so you can check:
```go
if f.NoFilters {
    return true
}
```